components/butterfly/src/server/mod.rs

```diff
@@ -744,6 +757,9 @@ impl Server {
                 {
                     // TODO (CM): Why are we inferring departure from
                     // a service rumor?
+
+                    // Purge "heat" information for a member that's gone
+                    self.rumor_heat.purge(&service_rumor.member_id);
```

Comment:
I wonder if this and the duplication on L804 could be avoided by putting this logic inside `insert_health_by_id`.

Reply:
I'd initially wanted to do that, too, but the `MemberList` knows nothing of the `RumorHeat` data structure 😦